File: sdk/src/main/java/org/coralibre/android/sdk/internal/datatypes/CapturedData.java
package org.coralibre.android.sdk.internal.datatypes;


public class CapturedData {


    /**
     *  Milliseconds since Epoch.
     */
    private final long captureTimestampMillis;

    /**
     * the 10-minute interval
     */
    private final ENInterval enInterval;

    /** in dB */
    private final byte rssi;

    /** 16 bytes */
    private final RollingProximityIdentifier rpi;

    /** 4 bytes */
    private final AssociatedEncryptedMetadata aem;


    public CapturedData(
        final long captureTimestamp,
        final byte rssi,
        final RollingProximityIdentifier rpi,
        final AssociatedEncryptedMetadata aem
    ) {
        this.captureTimestampMillis = captureTimestamp;
        this.enInterval = new ENInterval(captureTimestamp, true);
        this.rssi = rssi;
        this.rpi = rpi;
        this.aem = aem;
    }

    public Long getCaptureTimestampMillis() {
        return captureTimestampMillis;
    }

    public ENInterval getEnInterval() {
        return enInterval;
    }

    public byte getRssi() {
        return rssi;
    }

    public RollingProximityIdentifier getRpi() {
        return rpi;
    }

    public AssociatedEncryptedMetadata getAem() {
        return aem;
    }
}

File: sdk/src/main/java/org/coralibre/android/sdk/internal/datatypes/CapturedData.kt
package org.coralibre.android.sdk.internal.datatypes

class CapturedData {
    /**
     * Milliseconds since Epoch.
     */
    val captureTimestampMillis: Long

    /**
     * the 10-minute interval
     */
    val enInterval: ENInterval

    /** in dB  */
    val rssi: Byte

    /** 16 bytes  */
    val rpi: RollingProximityIdentifier

    /** 4 bytes  */
    val aem: AssociatedEncryptedMetadata

    constructor(
        captureTimestamp: Long,
        rssi: Byte,
        rpi: RollingProximityIdentifier,
        aem: AssociatedEncryptedMetadata
    ) {
        captureTimestampMillis = captureTimestamp
        enInterval = ENInterval(captureTimestamp, true)
        this.rssi = rssi
        this.rpi = rpi
        this.aem = aem
    }

    constructor(
        captureTimestamp: Long,
        rssi: Int,
        rpi: RollingProximityIdentifier,
        aem: AssociatedEncryptedMetadata
    ) {
        captureTimestampMillis = captureTimestamp
        enInterval = ENInterval(captureTimestamp, true)
        this.rssi = rssi.toByte()
        this.rpi = rpi
        this.aem = aem
    }

    val rssiInt: Int
        get() = rssi.toInt()
}
